community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/planner/CostBasedExecutablePlanBuilder.scala

```diff
@@ -69,35 +72,36 @@ case class CostBasedExecutablePlanBuilder(monitors: Monitors,
         val (logicalPlan, pipeBuildContext) = produceLogicalPlan(ast, rewrittenSemanticTable)(planContext)
         monitor.foundPlan(inputQuery.queryText, logicalPlan)
         try {
-          val res = produceCompiledPlan(logicalPlan, inputQuery, rewrittenSemanticTable,
-            planContext, pipeBuildContext, planBuilderMonitor)
+          val res = runtimeName match {
+            case CompiledRuntimeName => produceCompiledPlan(logicalPlan, rewrittenSemanticTable, planContext, planBuilderMonitor)
+            case _ => produceInterpretedPlan(logicalPlan, pipeBuildContext, planContext)
```

Comment:
This logic is completely broken, we should never fallback if you explicitly ask for `CompiledRuntimeName`, we need to fix the behavior here (which is btw not related to this PR).